messenger.go

```diff
@@ -435,6 +440,45 @@ func (m *Messenger) Attachment(to Recipient, dataType AttachmentType, url string
 	return response.Attachment(dataType, url, messagingType, tags...)
 }
 
+// EnableChatExtension set the homepage url required for a chat extension.
+func (m *Messenger) EnableChatExtension(homeURL HomeURL) error {
+	wrap := map[string]interface{}{
+		"home_url": homeURL,
+	}
+	data, err := json.Marshal(wrap)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", MessengerProfileURL, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.URL.RawQuery = "access_token=" + m.token
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	var respMsg map[string]*json.RawMessage
+	err = json.NewDecoder(resp.Body).Decode(&respMsg)
+	if err != nil {
+		return err
+	}
+
+	if errMsg, isErr := respMsg["error"]; isErr {
+		return fmt.Errorf("%s", errMsg)
```

Comment:
Fix here, on error message, the output look like

```
{
    "error": {
        "message": "",
    }
}
```

Reply:
Also "%s" take a string, and errMsg is a `json.RowMessage`,

Reply:
I would expect that the response you receive here matches the type `QueryResponse` defined in `response.go`:
https://github.com/paked/messenger/blob/e03d4b561e4f3130a44f154651681f855e9ea1e6/response.go#L50-L63

You should be able to use this type instead of a `map[string]*json.RawMessage` when decoding the response body.

Also, you can always use `%#v` instead of `%s` in your `Errorf` if you want to have more details on the data that you receive (see https://golang.org/pkg/fmt/#hdr-Printing), that's quite handy when debugging.

Reply:
Oh Awesome ! I didn't see the query response

I like to use `%#v` and use it all the time, but we should return only the message.

Reply:
squashed and updated